File: mobile-client/android/UseCase/Taxation/Detail/LoadTaxDetailUseCase.kt
package app.mobiling.client.client.usecase.taxation.detail

import app.mobiling.client.client.contract.taxation.detail.TaxDetailPayload
import app.mobiling.client.client.data.taxation.detail.TaxDetailGateway

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
class LoadTaxDetailUseCase(
    private val gateway: TaxDetailGateway,
) {
    suspend operator fun invoke(taxDocumentId: String): TaxDetailPayload = gateway.loadTaxDetail(taxDocumentId)
}

File: mobile-client/ios/UseCase/Taxation/Detail/LoadTaxDetailUseCase.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
struct LoadTaxDetailUseCase {
    let gateway: TaxDetailGateway

    func callAsFunction(taxDocumentId: String) async throws -> TaxDetailPayload {
        try await gateway.loadTaxDetail(taxDocumentId: taxDocumentId)
    }
}
